File: Low Level Design (LLD)/DesignPatterns/1. Creational/Builder/Cpp/src/AsusDesktopBuilder.cpp
#include "../include/AsusDesktopBuilder.hpp"

// Build the monitor
void AsusDesktopBuilder::buildMonitor() {
    desktop->setMonitor("Asus 24-inch 1080 144hz LED");
};

// Build the mouse
void AsusDesktopBuilder::buildMouse() {
    desktop->setMouse("Asus Gaming Mouse");
};

// Build the keyboard
void AsusDesktopBuilder::buildKeyboard() {
    desktop->setKeyboard("Asus Mechanical Keyboard");
};

// Build the speaker
void AsusDesktopBuilder::buildSpeaker() {
    desktop->setSpeaker("Asus SonicMaster");
};

// Build the RAM
void AsusDesktopBuilder::buildRam() {
    desktop->setRam("16GB DDR4");
};

// Build the processor
void AsusDesktopBuilder::buildProcessor() {
    desktop->setProcessor("AMD Ryzen 9");
};

// Build the motherboard
void AsusDesktopBuilder::buildMotherboard() {
    desktop->setMotherboard("Asus TUF 15");
};

// Build the graphics card
void AsusDesktopBuilder::buildGraphicsCard() {
    desktop->setGraphicsCard("Nvidia RTX 4050");
};

File: Low Level Design (LLD)/DesignPatterns/1. Creational/Builder/Java/AsusDesktopBuilder.java
public class AsusDesktopBuilder extends DesktopBuilder {

    @Override
    public void buildMonitor() {
        desktop.setMonitor("Asus 24-inch 1080 144hz LED");
    }

    @Override
    public void buildMouse() {
        desktop.setMouse("Asus Gaming Mouse");
    }

    @Override
    public void buildKeyboard() {
        desktop.setKeyboard("Asus Mechanical Keyboard");
    }

    @Override
    public void buildSpeaker() {
        desktop.setSpeaker("Asus SonicMaster");
    }

    @Override
    public void buildRam() {
        desktop.setRam("16GB DDR4");
    }

    @Override
    public void buildProcessor() {
        desktop.setProcessor("AMD Ryzen 9");
    }

    @Override
    public void buildMotherboard() {
        desktop.setMotherboard("Asus TUF 15");
    }

    @Override
    public void buildGraphicsCard() {
        desktop.setGraphicsCard("Nvidia RTX 4050");
    }
}
